tests/README.md

```diff
@@ -1,24 +1,100 @@
 # Integration tests
 
-Integration tests require a running Kubevirt cluster.  Once you have a running
-Kubevirt cluster, you can use the `-master` and the `-kubeconfig` flags to
-point the tests to the cluster.
+Kubevirt integration tests are a suite of end-to-end tests which run
+against an active Kubevirt cluster.
 
-## Run them on an arbitrary KubeVirt installation
+The tests may execute against various clusters, from integrated kubevirtci
+deployments and up to external kubernetes or OpenShift ones.
+
+## Kubevirtci
+
+Kubevirtci provides containerized Kubernetes clusters which can be used
+to test KubeVirt and similar projects.
 
+KubeVirt integration tests are being run against kubevirtci clusters on CI
+and is the simplest way to run the tests locally in a development environment.
+
+To run the test locally:
+- Configure the desired cluster setup (provider, number of nodes and nics).
+  Note: Some tests may run on on specific configuration and HW.
 ```
-cd tests # from the git repo root folder
-go test -kubeconfig=path/to/my/config -config=default-config.json
+export KUBEVIRT_PROVIDER=k8s-1.17
+export KUBEVIRT_NUM_NODES=2
+export KUBEVIRT_NUM_SECONDARY_NICS=2
 ```
 
-## Run them on one of the core KubeVirt providers
+- Raise a cluster based on the configuration.
+  At this stage, it will not contain kubevirt.
+  ```make cluster-up```
+- Build KubeVirt from sources.
+  ```make```
+- Deploy KubeVirt onto the cluster.
+  ```make cluster-sync```
+- Run the tests.
+  ```make functest```
+- Take down and delete the cluster.
+  ```make cluster-down```
+
+On a regular workflow of development or debugging, rebuilding KubeVirt is
+needed only when the production code is changed.
+In case only test code changes, `make functest` will rebuild the test code
+implicitly.
+
+## Existing (external) Kubevirt Cluster
+
+The tests may be executed against an already deployed (and active) Kubevirt
+cluster.
+
+The environment needs to be adjusted to allow the tests to operate agains
+such a cluster.
+The following high level steps are required to accomplish this:
+- Clone kubevirt repository on the machine that has access to the cluster.
+- Identify the deployed kubevirt version.
+- From the git repo, checkout to the commit that corrolates to the running
+kubevirt cluster.
+- Run the tests by executing the `run-tests-on-external-cluster.sh`.
+
+### Identify the commit on which the deployed kubevirt cluster is based on
 
-There is a make target to run this with the config
-taken from hack/config.sh:
+In order to identify on which commit the current kubevirt cluster is based on,
+there is a need to query a kubevirt infra pod and then inspect its base
+container.
+
+- Lookup for the `virt-operator` pod:
+  `kubectl get pods --all-namespaces | grep "virt-operator"`
```

Comment:
you can use `-A` instead of `--all-namespaces`

Reply:
Abbreviation in documentation or scripts is less common.
There is an interest to make it clear what it is, even for readers who are not familiar with it.

Reply:
i would write maybe a note about -A equal --all-namespaces ?
we are used to -A , so maybe it would worth to mention at least both side by side in a note

Reply:
Who is "we"? I think I was clear that in documentations abbreviations are not to be used as they are not readable and requires the reader to fetch the documentation help of the command.

Claiming that you know what -A means but not what --all-namespaces means is a surprise to me.

Reply:
its ok , i know both, but i suggest that a note there would also tell people who are new for it, that they can use -A
and dont overuse the keyboard

Reply:
This text is not about teaching ppl the optional shortcuts of arguments, it is about giving the instructions on how to use the system to reach the goal of doing a specific thing.

You should **not** use abbreviation in such documents and this is what is done here.

Reply:
i suggested to have -A as a note, because its the common favored practice, i dont mind (as usually unless i do mind),
but this is my opinion at least.